packages/block-editor/src/components/block-list/block.js

```diff
@@ -399,7 +398,7 @@ function BlockListBlock( {
 
 	return (
 		<animated.div
-			id={ blockElementId }
+			id={ clientId }
```

Comment:
It feels weird to me to have a DOM node with a hash as an id. 
Are we expecting any benefits from this PR (performance or something else)? I do wonder if it's just breaking existing behavior plugins could have relied on for no real gain?

Reply:
Why does it feel weird? And why does `block-{ clientId }` not feel weird? A client ID is unique per page so I don't see a problem with it.
It simplifies getting the client ID based on a DOM node. Currently we have to strip the `block-` prefix.
We don't consider element attributes to be part of an API right? I don't see why a plugin would use this, it supposed to be for internal use.

Reply:
We definitely don't consider this part of the API but it's something that plugin could have used; Why can't we just use the "data-block" attribute instead of stripping "block-"?

Reply:
Because a query to get a DOM node by client ID would be way more expensive then. `getElementById` is the fastest way to query the DOM. Since all blocks have a unique id, I don't see why we shouldn't take advantage of that. The `[data-block="{ clientId }"]` will need to go over every element in the DOM and check the attribute.